pkg/webui/console/containers/gateway-onboarding-form/index.js

```diff
@@ -86,37 +92,64 @@ const GatewayOnboardingForm = props => {
       const gatewayId = cleanValues.ids.gateway_id
 
       try {
-        await dispatch(attachPromise(createGateway(ownerId, cleanValues, isUserOwner)))
         if (_create_api_key_cups) {
-          await generateCUPSApiKey(gatewayId)
+          const generatedCupsKey = await generateCUPSApiKey(gatewayId)
+          setCupsKey(generatedCupsKey)
         }
         if (_create_api_key_lns) {
-          await generateLNSApiKey(gatewayId)
+          const generatedLnsKey = await generateLNSApiKey(gatewayId)
+          cleanValues.lbs_lns_secret = { value: generatedLnsKey }
+          setLnsKey(generatedLnsKey)
         }
+        await dispatch(attachPromise(createGateway(ownerId, cleanValues, isUserOwner)))
```

Comment:
You cannot create the gateway last. This will make the API key requests fail because the gateway does not exist yet.